### DISCUSSION TITLE
Incongruent behavior of SHOW/DESCRIBE?

### DISCUSSION THREAD

**Discussion by vlulla** [duckdb/duckdb#6393] [Q&A] [answered]
Following the example listed on the window function page I created a table, in a new empty duckdb database, with the following sql statement:

```sql
create table t(plant text, dt date, mwh bigint);
```

Running `show;` (or `describe;`) lists the following:
```
D show;
┌────────────┬──────────────────┬─────────────────────────┬───────────┐
│ table_name │   column_names   │      column_types       │ temporary │
│  varchar   │    varchar[]     │        varchar[]        │  boolean  │
├────────────┼──────────────────┼─────────────────────────┼───────────┤
│ t          │ [dt, mwh, plant] │ [DATE, BIGINT, VARCHAR] │ false     │
└────────────┴──────────────────┴─────────────────────────┴───────────┘
```

Whereas `show t;` displays:

```
D show t;
┌─────────────┬─────────────┬─────────┬─────────┬─────────┬───────┐
│ column_name │ column_type │  null   │   key   │ default │ extra │
│   varchar   │   varchar   │ varchar │ varchar │ varchar │ int32 │
├─────────────┼─────────────┼─────────┼─────────┼─────────┼───────┤
│ plant       │ VARCHAR     │ YES     │ Ø       │ Ø       │     Ø │
│ dt          │ DATE        │ YES     │ Ø       │ Ø       │     Ø │
│ mwh         │ BIGINT      │ YES     │ Ø       │ Ø       │     Ø │
└─────────────┴─────────────┴─────────┴─────────┴─────────┴───────┘
```

Notice how the output of `show` lists `[dt, mwh, plant]` whereas output of `show t` lists `plant`, `dt`, `mwh` (correct order as it was used in the `CREATE TABLE` statement!). It appears that the general `show` (or `describe`) statement is ordering the input by `column_name` whereas the specific `show t` (or `describe t`) is ordering by the `column_index`. Is this expected behavior?

More generally, can the output of `show` (or `describe`) be modified so that the output list of column names, and column types, is congruent with what would show up with a specific statement (`show t`)?

Thanks in advance!

**Comment by Mytherin** [comment, marked as answer]
Sure, I think that makes sense. I will push a PR.

**Reply by vlulla** [reply]
Thank you!
